lib/tool/src/tool_state.dart

```diff
@@ -12,7 +12,7 @@ class ToolState {
   static final provider = StateNotifierProvider<ToolStateNotifier, ToolState>(
     (ref) => ToolStateNotifier(
       ToolState(
-        currentTool: ref.watch(BrushTool.provider),
+        currentTool: ref.read(BrushTool.provider),
```

Comment:
Just for my understanding: Why do we just need to read it once here and not watch it?

Reply:
My idea was that it was not necessary to watch but apparently I was wrong. [When providers are combined we are supposed to always use watch.](https://riverpod.dev/docs/concepts/combining_providers) Changed read to watch where it was applicable.